src/main/java/org/opensearch/knn/indices/ModelCache.java

```diff
@@ -77,8 +80,9 @@ private void initCache() {
         CacheBuilder<String, Model> cacheBuilder = CacheBuilder.newBuilder()
                 .recordStats()
                 .concurrencyLevel(1)
-                .maximumWeight(cacheSizeInBytes)
-                .weigher((k, v) -> v.getLength());
+                .maximumWeight(cacheSizeInKB)
+                .expireAfterAccess(MODEL_CACHE_EXPIRE_AFTER_ACCESS_TIME_MINUTES, TimeUnit.MINUTES)
```

Comment:
What is the reasoning behind expiry time and should we increase to may be 3 HR to avoid frequent evictions and reload?